pkg/app/app.go

```diff
@@ -75,6 +77,23 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 		return errors.Wrap(err, "could not create runner output")
 	}
 	writer := metadata.NewWriter("kubetest2", junitRunner)
+
+	go func() {
+		// catch interrupt signals and gracefully attempt to clean up
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+		for range c {
+			if opts.ShouldUp() || opts.ShouldTest() {
+				klog.Info("Captured ^C, gracefully attempting to cleanup resources..")
+				if err := writer.WrapStep("Down", d.Down); err != nil {
```

Comment:
So currently we already have `down()` already as a defer so I imagine before this down() gets a chance to complete the deferred down() will be also invoked and the process would exit.

We can either fully rely on the deferred Down() and just `return` here or duplicate the existing Down() logic here and remove the defer.